test/common/router/header_parser_corpus/clusterfuzz-testcase-header_parser_fuzz_test-6195059702628352

```diff
@@ -0,0 +1,8 @@
+headers_to_add {
```

Comment:
@anirudhmurali can you add a test case to the unit tests for `header_formatter.cc`? Best practice is to check in both corpus and a unit test for the new error handling.